sdcm/nemesis.py

```diff
@@ -859,6 +860,94 @@ def disrupt_snapshot_operations(self):
+
+            READS Sampler:
+              Cardinality: ~10 (256 capacity)
+              Top 3 partitions:
+                Partition     Count       +/-
+                0         3         0
+                1         3         0
+                2         3         0
+                3         2         0
+
+            return Dict:
+            {
+                'READS': {
+                    'toppartitions': '10',
+                    'partitions': OrderedDict('0': {'count': '1', 'margin': '0'},
+                                              '1': {'count': '1', 'margin': '0'},
+                                              '2': {'count': '1', 'margin': '0'}),
+                    'cardinality': '10',
+                    'capacity': '256',
+                },
+                'WRITES': {
+                    'toppartitions': '10',
+                    'partitions': OrderedDict('10': {'count': '1', 'margin': '0'},
+                                              '11': {'count': '1', 'margin': '0'},
+                                              '21': {'count': '1', 'margin': '0'}),
+                    'cardinality': '10',
+                    'capacity': '256',
+                    'sampler': 'WRITES'
+                }
+            }
+
+
+            Arguments:
+                output {str} -- stdout of nodetool topparitions command
+
+            Returns:
+                dict -- result of parsing
+            """
+
+            pattern1 = "(?P<sampler>[A-Z]+)\sSampler:\W+Cardinality:\s~(?P<cardinality>[0-9]+)\s\((?P<capacity>[0-9]+)\scapacity\)\W+Top\s(?P<toppartitions>[0-9]+)\spartitions:"
+            pattern2 = "(?P<partition>[\w:]+)\s+(?P<count>[\d]+)\s+(?P<margin>[\d]+)"
+            toppartitions = {}
+            for out in output.split('\n\n'):
+                partition = OrderedDict()
+                sampler_data = re.match(pattern1, out, re.MULTILINE)
+                sampler_data = sampler_data.groupdict()
+                partitions = re.findall(pattern2, out, re.MULTILINE)
+                for v in partitions:
+                    partition.update({v[0]: {'count': v[1], 'margin': v[2]}})
+                sampler_data.update({'partitions': partition})
+                toppartitions[sampler_data.pop('sampler')] = sampler_data
+            return toppartitions
+
+        self._set_current_disruption("ShowTopPartitions")
+        self.target_node.run_nodetool('cfstats')
+        ks_cf_list = self.loaders.get_non_system_ks_cf_list(self.loaders.nodes[0], self.cluster.nodes[0])
+        top_partition = random.randint(5, 20)
+        samplers = random.choice(['writes', 'reads', 'writes,reads'])
+        capacity = random.randint(100, 1024)
+        ks, cf = random.choice(ks_cf_list).split('.')
+        duration = random.randint(1000, 10000)
+        cmd = "toppartitions {ks} {cf} {duration} -s {capacity} -k {top_partition} -a {samplers}"
+        result = self.target_node.run_nodetool(cmd.format(**locals()))
```

Comment:
use `sub_cmd="toppartitions"` and `args="{ks} {cf} {duration} -s {capacity} -k {top_partition} -a {samplers}"`

Reply:
fixed